code/parse/sexp.cpp

```diff
@@ -25197,6 +25198,59 @@ int sexp_is_in_mission(int node)
 	return SEXP_TRUE;
 }
 
+int sexp_has_armor_type(int node)
+{
+
+	// get ship from sexp
+	auto ship_entry = eval_ship(node);
+	if (!ship_entry || ship_entry->status == ShipStatus::NOT_YET_PRESENT)
+		return SEXP_NAN;
+	if (ship_entry->status == ShipStatus::EXITED)
+		return SEXP_NAN_FOREVER;
+
+	ship* shipp = ship_entry->shipp;
+	node = CDR(node);
+
+	// get armor from sexp
+	int armor;
+	if (!stricmp(SEXP_NONE_STRING, CTEXT(node))) {
+		armor = -1;
+	} else {
+		armor = armor_type_get_idx(CTEXT(node));
+
+		//warn and return if the armor is not found
+		if (armor < 0) {
+			Warning(LOCATION, "Armor %s not found on ship %s", CTEXT(node), shipp->ship_name);
```

Comment:
The warning message is not quite correct... the armor is not recognized as an armor type.  Yes, it's not found on the ship, but more fundamentally, it's not a valid entry in the armor table.

Reply:
done